// cydb/index.js

/*!
 * cydb
 * Copyright(c) 2022 Arda Demir DEMİRCİ
 * 
 * Mybutton Licensed
 */


/*!
Mybutton aşağıdaki şekilde kullanılmak üzere lisanslanmıştır:

"" "
Telif hakkı Mybutton katılımcıları. Tüm hakları Saklıdır.

Bu vesileyle, bir kopyasını alan herhangi bir kişiye ücretsiz olarak izin verilmektedir.
bu yazılımın ve ilgili belge dosyalarının ("Yazılım")
Yazılımda herhangi bir kısıtlama olmaksızın, bunlarla sınırlı olmamak üzere,
kullanma, kopyalama, değiştirme, birleştirme, yayınlama, dağıtma, alt lisanslama ve / veya
Yazılımın kopyalarını satmak ve Yazılımın bulunduğu kişilere izin vermek için
aşağıdaki koşullara tabi olarak temin edildi:

Yukarıdaki telif hakkı bildirimi ve bu izin bildirimi,
Yazılımın tüm kopyaları veya önemli kısımları.

YAZILIM AÇIK VEYA HERHANGİ BİR GARANTİ OLMAKSIZIN "OLDUĞU GİBİ" SAĞLANMAKTADIR.
TİCARİ ELVERİŞLİLİK GARANTİLERİ DAHİL ANCAK BUNLARLA SINIRLI OLMAMAK ÜZERE ZIMNİ,
BELİRLİ BİR AMACA UYGUNLUK VE İHLAL ETMEME. HİÇBİR DURUMDA
YAZARLAR VEYA TELİF HAKKI SAHİPLERİ HERHANGİ BİR TALEP, ZARAR VEYA DİĞER
SÖZLEŞME, HAKİM VEYA BAŞKA BİR ŞEKİLDE ORTAYA ÇIKAN SORUMLULUK
YAZILIM VEYA KULLANIM VEYA DİĞER ANLAŞMALARDAN, YAZILIMLA VEYA BAĞLANTILI OLARAK
YAZILIMDA.
"" "

Bu lisans, Mybutton'unn şuradan kaynaklanan bölümleri için geçerlidir:


"" "
Telif Hakkı internetiseviyorum ve diğer Düğüm katılımcıları. Tüm hakları Saklıdır.
Bu vesileyle, bir kopyasını alan herhangi bir kişiye ücretsiz olarak izin verilmektedir.
bu yazılımın ve ilgili belge dosyalarının ("Yazılım")
Yazılımda herhangi bir kısıtlama olmaksızın, bunlarla sınırlı olmamak üzere,
kullanma, kopyalama, değiştirme, birleştirme, yayınlama, dağıtma, alt lisanslama ve / veya
Yazılımın kopyalarını satmak ve Yazılımın bulunduğu kişilere izin vermek için
aşağıdaki koşullara tabi olarak temin edildi:

Yukarıdaki telif hakkı bildirimi ve bu izin bildirimi,
Yazılımın tüm kopyaları veya önemli kısımları.

YAZILIM AÇIK VEYA HERHANGİ BİR GARANTİ OLMAKSIZIN "OLDUĞU GİBİ" SAĞLANMAKTADIR.
TİCARİ ELVERİŞLİLİK GARANTİLERİ DAHİL ANCAK BUNLARLA SINIRLI OLMAMAK ÜZERE ZIMNİ,
BELİRLİ BİR AMACA UYGUNLUK VE İHLAL ETMEME. HİÇBİR DURUMDA
YAZARLAR VEYA TELİF HAKKI SAHİPLERİ HERHANGİ BİR TALEP, ZARAR VEYA DİĞER
SÖZLEŞME, TORT VEYA BAŞKA BİR ŞEKİLDE ORTAYA ÇIKAN SORUMLULUK
YAZILIM VEYA KULLANIM VEYA DİĞER ANLAŞMALARDAN, YAZILIMLA VEYA BAĞLANTILI OLARAK
YAZILIMDA.
"" "

Mybutton lisansı, harici olarak olmayan tüm Mybutton bölümleri için geçerlidir.
korunan kitaplıklar.

Mybutton tarafından kullanılan harici olarak tutulan kitaplıklar şunlardır:

- deps / acorn'da bulunan meşe palamudu aşağıdaki şekilde lisanslanmıştır:
  "" "
    MIT Lisansı

    Telif Hakkı (C) 2012-2018, çeşitli katkıda bulunanlar tarafından (bkz. YAZARLAR)

    Bu vesileyle, bir kopyasını alan herhangi bir kişiye ücretsiz olarak izin verilmektedir.
    bu yazılımın ve ilgili belge dosyalarının ("Yazılım")
    Yazılımda, haklar dahil ancak bunlarla sınırlı olmamak üzere, kısıtlama olmaksızın
    kullanmak, kopyalamak, değiştirmek, birleştirmek, yayınlamak, dağıtmak, alt lisans vermek ve / veya satmak
    Yazılımın kopyaları ve Yazılımın bulunduğu kişilere izin vermek için
    aşağıdaki koşullara tabi olarak temin edildi:

    Yukarıdaki telif hakkı bildirimi ve bu izin bildirimi,
    Yazılımın tüm kopyaları veya önemli kısımları.

    YAZILIM AÇIK VEYA HERHANGİ BİR GARANTİ OLMAKSIZIN "OLDUĞU GİBİ" SAĞLANMAKTADIR.
    TİCARİ ELVERİŞLİLİK GARANTİLERİ DAHİL ANCAK BUNLARLA SINIRLI OLMAMAK ÜZERE ZIMNİ,
    BELİRLİ BİR AMACA UYGUNLUK VE İHLAL ETMEME. HİÇBİR DURUMDA
    YAZARLAR VEYA TELİF HAKKI SAHİPLERİ HERHANGİ BİR TALEP, ZARAR VEYA DİĞER
    SÖZLEŞME, HAKSIZ VEYA BAŞKA ŞEKİLDE KAYNAKLANAN SORUMLULUK,
    YAZILIM VEYA KULLANIM YA DA DİĞER ANLAŞMALARLA BAĞLANTILI YA DA YAZILIMLA BAĞLANTILI
    YAZILIM.
  "" "

- deps / acorn-plugins'de bulunan Acorn eklentileri aşağıdaki şekilde lisanslanmıştır:
  "" "
    Telif Hakkı (C) 2017-2018, Adrian Heine

    Bu vesileyle, bir kopyasını alan herhangi bir kişiye ücretsiz olarak izin verilmektedir.
    bu yazılımın ve ilgili belge dosyalarının ("Yazılım")
    Yazılımda, haklar dahil ancak bunlarla sınırlı olmamak üzere, kısıtlama olmaksızın
    kullanmak, kopyalamak, değiştirmek, birleştirmek, yayınlamak, dağıtmak, alt lisans vermek ve / veya satmak
    Yazılımın kopyaları ve Yazılımın bulunduğu kişilere izin vermek için
    aşağıdaki koşullara tabi olarak temin edildi:

    Yukarıdaki telif hakkı bildirimi ve bu izin bildirimi,
    Yazılımın tüm kopyaları veya önemli kısımları.

    YAZILIM AÇIK VEYA HERHANGİ BİR GARANTİ OLMAKSIZIN "OLDUĞU GİBİ" SAĞLANMAKTADIR.
    TİCARİ ELVERİŞLİLİK GARANTİLERİ DAHİL ANCAK BUNLARLA SINIRLI OLMAMAK ÜZERE ZIMNİ,
    BELİRLİ BİR AMACA UYGUNLUK VE İHLAL ETMEME. HİÇBİR DURUMDA
    YAZARLAR VEYA TELİF HAKKI SAHİPLERİ HERHANGİ BİR TALEP, ZARAR VEYA DİĞER
    SÖZLEŞME, HAKSIZ VEYA BAŞKA ŞEKİLDE KAYNAKLANAN SORUMLULUK,
    YAZILIM VEYA KULLANIM YA DA DİĞER ANLAŞMALARLA BAĞLANTILI YA DA YAZILIMLA BAĞLANTILI
    YAZILIM.
  "" "

- deps / care'de bulunan c-ares, aşağıdaki şekilde lisanslanmıştır:
*/

/**
 * Module exports.
 */

const fs = require("fs")
const { dirname } = require("path")
const chalk = require("chalk")

class DB {

    constructor(){

        
if (!fs.existsSync('./CYdatabase.json')) {
 fs.writeFile('CYdatabase.json', '{\n"config": true\n}\n', function err(err) {
    if (err){
        throw new TypeError(chalk.red(err))
    }
    })
    }
}

/*    CYDB TÜRKÇE VERSİYONU                                                                                                 */
    yaz(veri, değer){
        
        if(!veri) throw new TypeError(chalk.red("Veri Girmediniz "))
        if(!değer) throw new TypeError(chalk.red("Değer Girmediniz "))
        const dosya = JSON.parse(fs.readFileSync("CYdatabase.json", "utf8"))
        dosya[veri] = değer 
        return fs.writeFileSync("CYdatabase.json", JSON.stringify(dosya, null, 2))
    }//yazı
    bul(veri){
        if(!veri) throw new TypeError(chalk.red("Veri Girmediniz "))
    const dosya = JSON.parse(fs.readFileSync("CYdatabase.json", "utf8"))
    if(!dosya[veri]) throw new TypeError(chalk.yellow("CYDB (Cyber Database) Yazdığınız Veriyi Bulamadı \n" + __dirname))
    return dosya[veri]
    }//okuma

    kontrol(veri){
        if(!veri) throw new TypeError(chalk.red("Veri Girmediniz "))
        const dosya = JSON.parse(fs.readFileSync("CYdatabase.json", "utf8"))
        return dosya [veri] ? true : false 
    }//kontrol
    sil(veri){
        if(!veri) throw new TypeError(chalk.red("Veri Girmediniz "))
        const dosya = JSON.parse(fs.readFileSync("CYdatabase.json", "utf8"))
        if(!dosya[veri]) throw new TypeError(chalk.yellow("CYDB (Cyber Database) Yazdığınız Veriyi Bulamadı \n" + __dirname))
        delete dosya[veri]
        return fs.writeFileSync("CYdatabase.json", JSON.stringify(dosya, null, 2))
    }//sil 
    yedekle(dosyaadı){
        if(!dosyaadı) throw TypeError(chalk.red("Dosya Adı  Girmediniz "))
        const dosya = JSON.parse(fs.readFileSync("CYdatabase.json", "utf8"))
        return fs.writeFileSync(`${dosyaadı}.json`, JSON.stringify(dosya , null , 2 ))
    }//yedekleme

    topla(veri, değer){
        if(!veri) throw new TypeError(chalk.red("Veri Girmediniz "))
        if(typeof değer !== "number") throw new TypeError(chalk.blue("Değer Olarak Sayı Giriniz, Girmediniz "))
        if(!this.kontrol(veri)) throw new TypeError(chalk.red("Veri Olarak Girdiğiniz Değer Bulunamadı  "))
        if(typeof this.bul(veri) !== "number") throw new TypeError(chalk.bold("Sayı Ekleyeceğiniz Değer Sayı Olmalı "))
        const dosya = JSON.parse(fs.readFileSync("CYdatabase.json", "utf8"))
        dosya[veri] += değer
        return fs.writeFileSync("CYdatabase.json", JSON.stringify(dosya, null, 2))
    }//toplama

    çıkar(veri,değer){
        if(!veri) throw new TypeError(chalk.red("Veri Girmediniz "))
        if(typeof değer !== "number") throw new TypeError(chalk.blue("Değer Olarak Sayı Giriniz, Girmediniz "))
        if(!this.kontrol(veri)) throw new TypeError(chalk.red("Veri Olarak Girdiğiniz Değer Bulunamadı  "))
        if(typeof this.bul(veri) !== "number") throw new TypeError(chalk.bold("Sayı Ekleyeceğiniz Değer Sayı Olmalı "))
        const dosya = JSON.parse(fs.readFileSync("CYdatabase.json", "utf8"))
        dosya[veri] -= değer
        return fs.writeFileSync("CYdatabase.json", JSON.stringify(dosya, null, 2))
    }//çıkarma
    sıfırla(){
        const dosya = JSON.parse(fs.readFileSync("CYdatabase.json", "utf8"))
        return fs.writeFileSync("CYdatabase.json", JSON.stringify({}, null , 2 ))
      
    }//veri sıfırlama

   depola(veri, değer) {
        if (!veri) throw new TypeError("Bir veri belirtilmedi!");
        if (!değer) throw new TypeError("Bir değer belirtilmedi!");
        let dosya = JSON.parse(fs.readFileSync("CYdatabase.json", "utf8"));
        if (dosya[veri] && veri.isveri(dosya[veri])) {
          dosya[veri].push(değer)
          fs.writeFileSync("CYdatabase.json", JSON.stringify(dosya, null, 1))
        } else if (!dosya[veri]) {
          this.yaz(veri, [değer])
        }
        return
      }
   

/*         CYDB ENGLİSH VERSİON                                                                                               */
    
set(veri, değer){
        
    if(!veri) throw new TypeError(chalk.red("You Didn't Enter Data"))
    if(!değer) throw new TypeError(chalk.red("You Didn't Enter a Value"))
    const dosya = JSON.parse(fs.readFileSync("CYdatabase.json", "utf8"))
    dosya[veri] = değer 
    return fs.writeFileSync("CYdatabase.json", JSON.stringify(dosya, null, 2))
}//yazı
get(veri){
    if(!veri) throw new TypeError(chalk.red("You Didn't Enter Data "))
const dosya = JSON.parse(fs.readFileSync("CYdatabase.json", "utf8"))
if(!dosya[veri]) throw new TypeError(chalk.yellow(" CYDB (Cyber ​​Database)Couldn't Find The Data You Typed \n" + __dirname))
return dosya[veri]
}//okuma

fetch(veri){
    if(!veri) throw new TypeError(chalk.red("You Didn't Enter Data "))
const dosya = JSON.parse(fs.readFileSync("CYdatabase.json", "utf8"))
if(!dosya[veri]) throw new TypeError(chalk.yellow(" CYDB (Cyber ​​Database)Couldn't Find The Data You Typed \n" + __dirname))
return dosya[veri]
}//okuma

check(veri){
    if(!veri) throw new TypeError(chalk.red("You Didn't Enter Data"))
    const dosya = JSON.parse(fs.readFileSync("CYdatabase.json", "utf8"))
    return dosya [veri] ? true : false 
}//kontrol
has(veri){
    if(!veri) throw new TypeError(chalk.red("You Didn't Enter Data"))
    const dosya = JSON.parse(fs.readFileSync("CYdatabase.json", "utf8"))
    return dosya [veri] ? true : false 
}//kontrol
delete(veri){
    if(!veri) throw new TypeError(chalk.red("You Didn't Enter Data"))
    const dosya = JSON.parse(fs.readFileSync("CYdatabase.json", "utf8"))
    if(!dosya[veri]) throw new TypeError(chalk.yellow("Cyber Database Yazdığınız Veriyi Bulamadı \n" + __dirname))
    delete dosya[veri]
    return fs.writeFileSync("CYdatabase.json", JSON.stringify(dosya, null, 2))
}//sil 
backup(dosyaadı){
    if(!dosyaadı) throw TypeError(chalk.red("You Didn't Enter a File Name"))
    const dosya = JSON.parse(fs.readFileSync("CYdatabase.json", "utf8"))
    return fs.writeFileSync(`${dosyaadı}.json`, JSON.stringify(dosya , null , 2 ))
}//yedekleme

add(veri, değer){
    if(!veri) throw new TypeError(chalk.red("You Didn't Enter Data"))
    if(typeof değer !== "number") throw new TypeError(chalk.blue("Enter Number as Value You Didn't Enter "))
    if(!this.kontrol(veri)) throw new TypeError(chalk.red("The Value You Entered As Data Could Not Be Found  "))
    if(typeof this.bul(veri) !== "number") throw new TypeError(chalk.bold("Sayı Ekleyeceğiniz Değer Sayı Olmalı "))
    const dosya = JSON.parse(fs.readFileSync("CYdatabase.json", "utf8"))
    dosya[veri] += değer
    return fs.writeFileSync("CYdatabase.json", JSON.stringify(dosya, null, 2))
}//toplama

substr(veri,değer){
    if(!veri) throw new TypeError(chalk.red("You Didn't Enter Data "))
    if(typeof değer !== "number") throw new TypeError(chalk.blue("Enter Number as Value You Didn't Enter"))
    if(!this.kontrol(veri)) throw new TypeError(chalk.red("The Value You Entered As Data Could Not Be Found "))
    if(typeof this.bul(veri) !== "number") throw new TypeError(chalk.bold("Sayı Ekleyeceğiniz Değer Sayı Olmalı "))
    const dosya = JSON.parse(fs.readFileSync("CYdatabase.json", "utf8"))
    dosya[veri] -= değer
    return fs.writeFileSync("CYdatabase.json", JSON.stringify(dosya, null, 2))
}//çıkarma
reset(){
    const dosya = JSON.parse(fs.readFileSync("CYdatabase.json", "utf8"))
    return fs.writeFileSync("CYdatabase.json", JSON.stringify({}, null , 2 ))
  
}//veri sıfırlama



store(veri, değer) {
  if (!veri) throw new TypeError("You Didn't Enter Data ");
  if (!değer) throw new TypeError("No value specified.");
  let dosya = JSON.parse(fs.readFileSync("CYdatabase.json", "utf8"));
  if (dosya[veri] && veri.isveri(dosya[veri])) {
    dosya[veri].push(değer)
    fs.writeFileSync("CYdatabase.json", JSON.stringify(dosya, null, 1))
  } else if (!dosya[veri]) {
    this.yaz(veri, [değer])
  }
  return
}
}


/**
 * Mybutton aşağıdaki şekilde kullanılmak üzere lisanslanmıştır:

"" "
Telif hakkı Mybutton katılımcıları. Tüm hakları Saklıdır.

Bu vesileyle, bir kopyasını alan herhangi bir kişiye ücretsiz olarak izin verilmektedir.
bu yazılımın ve ilgili belge dosyalarının ("Yazılım")
Yazılımda herhangi bir kısıtlama olmaksızın, bunlarla sınırlı olmamak üzere,
kullanma, kopyalama, değiştirme, birleştirme, yayınlama, dağıtma, alt lisanslama ve / veya
Yazılımın kopyalarını satmak ve Yazılımın bulunduğu kişilere izin vermek için
aşağıdaki koşullara tabi olarak temin edildi:

Yukarıdaki telif hakkı bildirimi ve bu izin bildirimi,
Yazılımın tüm kopyaları veya önemli kısımları.

YAZILIM AÇIK VEYA HERHANGİ BİR GARANTİ OLMAKSIZIN "OLDUĞU GİBİ" SAĞLANMAKTADIR.
TİCARİ ELVERİŞLİLİK GARANTİLERİ DAHİL ANCAK BUNLARLA SINIRLI OLMAMAK ÜZERE ZIMNİ,
BELİRLİ BİR AMACA UYGUNLUK VE İHLAL ETMEME. HİÇBİR DURUMDA
YAZARLAR VEYA TELİF HAKKI SAHİPLERİ HERHANGİ BİR TALEP, ZARAR VEYA DİĞER
SÖZLEŞME, HAKİM VEYA BAŞKA BİR ŞEKİLDE ORTAYA ÇIKAN SORUMLULUK
YAZILIM VEYA KULLANIM VEYA DİĞER ANLAŞMALARDAN, YAZILIMLA VEYA BAĞLANTILI OLARAK
YAZILIMDA.
"" "

Bu lisans, Mybutton'unn şuradan kaynaklanan bölümleri için geçerlidir:


"" "
Telif Hakkı internetiseviyorum ve diğer Düğüm katılımcıları. Tüm hakları Saklıdır.
Bu vesileyle, bir kopyasını alan herhangi bir kişiye ücretsiz olarak izin verilmektedir.
bu yazılımın ve ilgili belge dosyalarının ("Yazılım")
Yazılımda herhangi bir kısıtlama olmaksızın, bunlarla sınırlı olmamak üzere,
kullanma, kopyalama, değiştirme, birleştirme, yayınlama, dağıtma, alt lisanslama ve / veya
Yazılımın kopyalarını satmak ve Yazılımın bulunduğu kişilere izin vermek için
aşağıdaki koşullara tabi olarak temin edildi:

Yukarıdaki telif hakkı bildirimi ve bu izin bildirimi,
Yazılımın tüm kopyaları veya önemli kısımları.

YAZILIM AÇIK VEYA HERHANGİ BİR GARANTİ OLMAKSIZIN "OLDUĞU GİBİ" SAĞLANMAKTADIR.
TİCARİ ELVERİŞLİLİK GARANTİLERİ DAHİL ANCAK BUNLARLA SINIRLI OLMAMAK ÜZERE ZIMNİ,
BELİRLİ BİR AMACA UYGUNLUK VE İHLAL ETMEME. HİÇBİR DURUMDA
YAZARLAR VEYA TELİF HAKKI SAHİPLERİ HERHANGİ BİR TALEP, ZARAR VEYA DİĞER
SÖZLEŞME, TORT VEYA BAŞKA BİR ŞEKİLDE ORTAYA ÇIKAN SORUMLULUK
YAZILIM VEYA KULLANIM VEYA DİĞER ANLAŞMALARDAN, YAZILIMLA VEYA BAĞLANTILI OLARAK
YAZILIMDA.
"" "

Mybutton lisansı, harici olarak olmayan tüm Mybutton bölümleri için geçerlidir.
korunan kitaplıklar.

Mybutton tarafından kullanılan harici olarak tutulan kitaplıklar şunlardır:

- deps / acorn'da bulunan meşe palamudu aşağıdaki şekilde lisanslanmıştır:
  "" "
    MIT Lisansı

    Telif Hakkı (C) 2012-2018, çeşitli katkıda bulunanlar tarafından (bkz. YAZARLAR)

    Bu vesileyle, bir kopyasını alan herhangi bir kişiye ücretsiz olarak izin verilmektedir.
    bu yazılımın ve ilgili belge dosyalarının ("Yazılım")
    Yazılımda, haklar dahil ancak bunlarla sınırlı olmamak üzere, kısıtlama olmaksızın
    kullanmak, kopyalamak, değiştirmek, birleştirmek, yayınlamak, dağıtmak, alt lisans vermek ve / veya satmak
    Yazılımın kopyaları ve Yazılımın bulunduğu kişilere izin vermek için
    aşağıdaki koşullara tabi olarak temin edildi:

    Yukarıdaki telif hakkı bildirimi ve bu izin bildirimi,
    Yazılımın tüm kopyaları veya önemli kısımları.

    YAZILIM AÇIK VEYA HERHANGİ BİR GARANTİ OLMAKSIZIN "OLDUĞU GİBİ" SAĞLANMAKTADIR.
    TİCARİ ELVERİŞLİLİK GARANTİLERİ DAHİL ANCAK BUNLARLA SINIRLI OLMAMAK ÜZERE ZIMNİ,
    BELİRLİ BİR AMACA UYGUNLUK VE İHLAL ETMEME. HİÇBİR DURUMDA
    YAZARLAR VEYA TELİF HAKKI SAHİPLERİ HERHANGİ BİR TALEP, ZARAR VEYA DİĞER
    SÖZLEŞME, HAKSIZ VEYA BAŞKA ŞEKİLDE KAYNAKLANAN SORUMLULUK,
    YAZILIM VEYA KULLANIM YA DA DİĞER ANLAŞMALARLA BAĞLANTILI YA DA YAZILIMLA BAĞLANTILI
    YAZILIM.
  "" "

- deps / acorn-plugins'de bulunan Acorn eklentileri aşağıdaki şekilde lisanslanmıştır:
  "" "
    Telif Hakkı (C) 2017-2018, Adrian Heine

    Bu vesileyle, bir kopyasını alan herhangi bir kişiye ücretsiz olarak izin verilmektedir.
    bu yazılımın ve ilgili belge dosyalarının ("Yazılım")
    Yazılımda, haklar dahil ancak bunlarla sınırlı olmamak üzere, kısıtlama olmaksızın
    kullanmak, kopyalamak, değiştirmek, birleştirmek, yayınlamak, dağıtmak, alt lisans vermek ve / veya satmak
    Yazılımın kopyaları ve Yazılımın bulunduğu kişilere izin vermek için
    aşağıdaki koşullara tabi olarak temin edildi:

    Yukarıdaki telif hakkı bildirimi ve bu izin bildirimi,
    Yazılımın tüm kopyaları veya önemli kısımları.

    YAZILIM AÇIK VEYA HERHANGİ BİR GARANTİ OLMAKSIZIN "OLDUĞU GİBİ" SAĞLANMAKTADIR.
    TİCARİ ELVERİŞLİLİK GARANTİLERİ DAHİL ANCAK BUNLARLA SINIRLI OLMAMAK ÜZERE ZIMNİ,
    BELİRLİ BİR AMACA UYGUNLUK VE İHLAL ETMEME. HİÇBİR DURUMDA
    YAZARLAR VEYA TELİF HAKKI SAHİPLERİ HERHANGİ BİR TALEP, ZARAR VEYA DİĞER
    SÖZLEŞME, HAKSIZ VEYA BAŞKA ŞEKİLDE KAYNAKLANAN SORUMLULUK,
    YAZILIM VEYA KULLANIM YA DA DİĞER ANLAŞMALARLA BAĞLANTILI YA DA YAZILIMLA BAĞLANTILI
    YAZILIM.
  "" "

- deps / care'de bulunan c-ares, aşağıdaki şekilde lisanslanmıştır:
 */
module.exports = new DB()
